Timed out, unknown build status on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.39125&attempt_id=a608d6c3-7d55-4de9-8e8d-cf2509038882)

Attempted: libharu, libiodbc, saga, saga_2_3_2, vigra

<details><summary>Partial log (click to expand)</summary><p>

```
       ^~~~
/build/hdf-4.2.14/hdf/src/hfile.h:302:7: error: unknown type name 'intn'
       intn       an_num[4];   /* Holds number of annotations found of each type */
       ^~~~
/build/hdf-4.2.14/hdf/src/hfile.h:277:7: error: unknown type name 'intn'
building of '/nix/store/nq55f8bi0mrqj4b0gixdw633ycfv606a-hdf-4.2.14.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/k2bx54rlidrmsaq1v0ab1s74j9y0z67m-gdal-2.3.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/3160ik6jgfb8wbswjjq3wqsi7g51ixbx-saga-2.3.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/xhrf36l2zb0kdfdzzq27wz7yzzvxrhzv-saga-6.3.0.drv': 1 dependencies couldn't be built
error: build of '/nix/store/3160ik6jgfb8wbswjjq3wqsi7g51ixbx-saga-2.3.2.drv', '/nix/store/xhrf36l2zb0kdfdzzq27wz7yzzvxrhzv-saga-6.3.0.drv' failed
```
</p></details>

--- a/pkgs/applications/gis/saga/default.nix
+++ b/pkgs/applications/gis/saga/default.nix
@@ -1,17 +1,22 @@
 { stdenv, fetchurl, gdal, wxGTK30, proj, libiodbc, lzma, jasper,
   libharu, opencv, vigra, postgresql, Cocoa,
-  unixODBC , poppler, hdf4, hdf5, netcdf, sqlite, qhull }:
+  unixODBC , poppler, hdf4, hdf5, netcdf, sqlite, qhull, giflib }:
 
 stdenv.mkDerivation rec {
   name = "saga-6.3.0";
 
   # See https://groups.google.com/forum/#!topic/nix-devel/h_vSzEJAPXs
   # for why the have additional buildInputs on darwin
-  buildInputs = [ gdal wxGTK30 proj libharu opencv vigra postgresql libiodbc lzma jasper qhull ]
+  buildInputs = [ gdal wxGTK30 proj libharu opencv vigra postgresql libiodbc lzma
+                  jasper qhull giflib ]
                 ++ stdenv.lib.optionals stdenv.isDarwin
                   [ Cocoa unixODBC poppler hdf4.out hdf5 netcdf sqlite ];
 
   enableParallelBuilding = true;
+
+  patches = [ ./finite-6.3.0.patch];
+
+  CXXFLAGS = stdenv.lib.optionalString stdenv.cc.isClang "-std=c++11 -Wno-narrowing";
 
   src = fetchurl {
     url = "mirror://sourceforge/project/saga-gis/SAGA%20-%206/SAGA%20-%206.3.0/saga-6.3.0.tar.gz";

--- a/pkgs/applications/gis/saga/finite-6.3.0.patch
+++ b/pkgs/applications/gis/saga/finite-6.3.0.patch
@@ -1,0 +1,55 @@
+diff --git a/src/tools/imagery/imagery_maxent/me.cpp b/src/tools/imagery/imagery_maxent/me.cpp
+index c5da854..d3e9cff 100755
+--- a/src/tools/imagery/imagery_maxent/me.cpp
++++ b/src/tools/imagery/imagery_maxent/me.cpp
+@@ -21,7 +21,7 @@
+ #ifdef _SAGA_MSW
+ #define isinf(x) (!_finite(x))
+ #else
+-#define isinf(x) (!finite(x))
++#define isinf(x) (!isfinite(x))
+ #endif
+ 
+ /** The input array contains a set of log probabilities lp1, lp2, lp3
+@@ -47,7 +47,7 @@ double sumLogProb(vector<double>& logprobs)
+ /** returns log (e^logprob1 + e^logprob2). */
+ double sumLogProb(double logprob1, double logprob2)
+ {
+-  if (isinf(logprob1) && isinf(logprob2)) 
++  if (isinf(logprob1) && isinf(logprob2))
+     return logprob1; // both prob1 and prob2 are 0, return log 0.
+   if (logprob1>logprob2)
+     return logprob1+log(1+exp(logprob2-logprob1));
+@@ -70,8 +70,8 @@ void MaxEntModel::print(ostream& ostrm, MaxEntTrainer& trainer)
+   for (FtMap::iterator it = _index.begin(); it!=_index.end(); it++) {
+     unsigned long i = it->second;
+     for (unsigned long c = 0; c<_classes; c++) {
+-      ostrm << "lambda(" << trainer.className(c) << ", " 
+-	    << trainer.getStr(it->first) << ")=" 
++      ostrm << "lambda(" << trainer.className(c) << ", "
++	    << trainer.getStr(it->first) << ")="
+ 	    << _lambda[i+c] << endl;
+     }
+   }
+@@ -86,7 +86,7 @@ int MaxEntModel::getProbs(MaxEntEvent& event, vector<double>& probs)
+     double s = 0;
+     for (unsigned int f = 0; f<event.size(); f++) {
+       FtMap::iterator it = _index.find(event[f]);
+-      if (it!=_index.end()) 
++      if (it!=_index.end())
+ 	s += _lambda[it->second+c];
+     }
+     probs[c] = s;
+@@ -142,10 +142,10 @@ double MaxEntModel::getObsCounts(EventSet& events, vector<double>& obsCounts)
+     double ftSum = 0;
+     for (unsigned long j=0; j<e.size(); j++) {
+       FtMap::iterator it = _index.find(e[j]);
+-      if (it!=_index.end()) 
++      if (it!=_index.end())
+ 	obsCounts[it->second+c] += count;
+       else { // new feature, need to expand obsCounts and _lambda
+-	for (unsigned int k = 0; k<_classes; k++) 
++	for (unsigned int k = 0; k<_classes; k++)
+ 	  obsCounts.push_back(0);
+ 	obsCounts[_lambda.size()+c] += count;
+ 	addFeature(e[j]);

--- a/pkgs/applications/gis/saga/lts.nix
+++ b/pkgs/applications/gis/saga/lts.nix
@@ -14,8 +14,6 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  GDAL_CFLAGS = "-std=c++11";
-
   CXXFLAGS = stdenv.lib.optionalString stdenv.cc.isClang "-std=c++11 -Wno-narrowing";
 
   sourceRoot = "code-b6f474f/saga-gis";

